### ISSUE TITLE
IP address on control board

### ISSUE THREAD

**Issue by levo-man** [PiSCSI/piscsi#1096]
# Info
Looking for a way to display the IP address of the Pi with the control board software.

- Which version of Pi are you using:
- Which github revision of software:
- Which board version:
- Which computer is the PiSCSI connected to:

# Describe the issue
Looking for a way to display the IP address of the Pi with the control board software. Either as a menu item , or at the bottom of the list of drives ( like the OLED software does ) . 
If as a menu item would it be possible to show the IP for any interface that is connected wlan0 or eth0 but only if it has an IP assigned.

**Comment by rdmark** [comment]
@bzeiss Is this something that you had on your roadmap?

**Comment by bzeiss** [comment]
It's a good idea. However, before working on new features, I want to check whether the 'break-in' problem of the rotary is something that can be solved. I suspect it's a problem inherent to the fact that the rotary goes through the multiplexer, usage of python for timing sensitive events as a rather slow language and the raspberry pi os doing a lot of other things besides looking for rotary encoder events.

**Comment by uweseimet** [comment]
@bzeiss I'm looking for stale tickets and have stumbled upon this one. Any news on this?

**Comment by levo-man** [comment]
I don't know about 'stale' but it is a project that is still available 
for people to build .



------ Original Message ------
From "Uwe Seimet" ***@***.***>
To "PiSCSI/piscsi" ***@***.***>
Cc "levo-man" ***@***.***>; "Author" 
***@***.***>
Date 16/10/2023 22:56:52
Subject Re: [PiSCSI/piscsi] IP address on control board (Issue #1096)

**Comment by uweseimet** [comment]
@levo-man Sure, but I was wondering whether @bzeiss has an updated status. Nobody else appears to be involved in this ticket.

**Comment by bzeiss** [comment]
Yes, fair question. My current impression is that there is little interest in the control board and I'm not sure whether it's worth spending a lot of time investigating a problem that is probably better solved with a different hardware design. Rotarys over I2C multiplexer are probably just not a very good idea at the end of the day (especially if the operating system is doing different things like scsi emulation at the same time). To be honest, we should probably think about removing the control board code.
@akuker @rdmark

**Comment by levo-man** [comment]
Or , maybe just do a redesign of the control board . I do like the 
functionality of it and will be doing a blog on a build soon .


------ Original Message ------
From "Benjamin Zeiss" ***@***.***>
To "PiSCSI/piscsi" ***@***.***>
Cc "levo-man" ***@***.***>; "Mention" 
***@***.***>
Date 17/10/2023 09:13:18
Subject Re: [PiSCSI/piscsi] IP address on control board (Issue #1096)

**Comment by akuker** [comment]
I don't think the control board is dead. There haven't been many recent updates, because its relatively simple and works.

A re-design of the controller board would be good. It should have its own microcontroller to handle the rotary encoder so the Pi doesn't have to deal with it.

**Comment by levo-man** [comment]
Yes I think tha that might be the way to go , if a rotary is definitely 
needed . The trouble is that I am stacked with things to repair , and 
projects to do at the moment .
Something I would definitely be willing to look at in the future .

alan .


------ Original Message ------
From "Tony Kuker" ***@***.***>
To "PiSCSI/piscsi" ***@***.***>
Cc "levo-man" ***@***.***>; "Mention" 
***@***.***>
Date 18/10/2023 02:47:14
Subject Re: [PiSCSI/piscsi] IP address on control board (Issue #1096)

**Comment by rdmark** [comment]
feature added in v25.12.01

**Comment by levo-man** [comment]
I will test as soon as the weather warms up a bit and I can get the A500+ on the bench with the A590 .

**Comment by rdmark** [comment]
FTR, this is how I designed it. You go to the menu for any SCSI ID, and then scroll to the bottom where you can find a **System Info** menu item.

![Image](https://github.com/user-attachments/assets/154587cd-acd4-40b1-a558-9adb6fb3b53d)
![Image](https://github.com/user-attachments/assets/4373f9b6-18fe-4e33-8f49-21879cf4fd6a)
